### ISSUE TITLE
rancher Kubernetes unable to tty/shell/ read logs in machines

### ISSUE THREAD

**Issue by lilnate22** [rancher/rancher#5949]
**Rancher Version:**
1.1.3
**Docker Version:**
1.10.3
**OS and where are the hosts located? (cloud, bare metal, etc):**
AWS ubuntu
**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
single node rancher, rancher used to create 2 hosts via AWS keys.
**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
kubernetes
**Steps to Reproduce:**
follow this guide: http://kubernetes.io/docs/user-guide/horizontal-pod-autoscaling/walkthrough/

when you get to step to tty to the pod...it will error
**Results:** 
TTY connection is refused, in my case with error: 

```
Error attaching, falling back to logs: dial tcp: lookup kubernetes-poc1: no such host
Error from server: Get https://kubernetes-poc1:10250/containerLogs/default/load-generator-1537631235-jedrt/load-generator: dial tcp: lookup kubernetes-poc1 on 169.254.169.250:53: no such host
```

I have no idea where the 169.X IP is coming from, since neither my AWS instances has that ip as external/internal.

Attempt to use the rancher UI to get shell access to a pod. Observe that the shell screen will instantly close. likely due to above error
**Expected:**
you should be able to access the shell of a pod.

**Comment by aemneina** [comment]
hey @lilnate22 did you manage to get to the bottom of this? Looks like kubectl can't resolve kubernetes-poc1 as a hostname, that would strike me as the first issue. That 169 address looks like the rancher internal DNS. Where are you trying to execute the kubectl exec or logs from?

**Comment by lilnate22** [comment]
@aemneina not really...im not actively using rancher k8s right now (waiting on rancher 1.2)

but kubectl was run from both my machine, as well as the ranchers console k8s

**Comment by wlan0** [comment]
The hostname doesn't seem resolvable. Kubectl makes a call to apiserver and apiserver proxies the call to the appropriate kubelet. 

Can you ensure that you can ping that host from the host that is running API-server? Let me know if you can.

**Comment by aemneina** [comment]
@lilnate22 cool, I'll close this out. If you ever try this out again, and hit issues, feel free to open a new ticket.
